File: ruby/robot-name/robot.rb
# Robot
class Robot
  attr_reader :name
  def initialize
    reset
  end

  def reset
    @name = generate_name
  end

  private

  def generate_name
    prefix + suffix
  end

  def prefix
    (0...2).map { ('A'..'Z').to_a[rand(26)] }.join
  end

  def suffix
    (0...3).map { ('1'..'9').to_a[rand(9)] }.join
  end
end

File: python/robot-name/robot.py
import string
import random


class Robot:

    def __init__(self):
        self.name = self.generate_name()

    def reset(self):
        self.name = self.generate_name()

    def generate_name(self):
        random.seed()
        return self.random_prefix(2) + self.random_suffix(3)

    def random_prefix(self, n):
        return ''.join(random.sample(string.ascii_uppercase, n))

    def random_suffix(self, n):
        return ''.join(random.sample(string.digits, n))
